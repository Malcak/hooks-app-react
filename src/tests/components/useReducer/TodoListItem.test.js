import React from 'react';
import { shallow } from 'enzyme';
import { TodoListItem } from '../../../components/useReducer/TodoListItem';
import { defaultTodos } from '../../fixtures/defaultTodos';

describe('tests on the <TodoListItem /> component', () => {
  const index = 0;
  const handleDelete = jest.fn();
  const handleToggle = jest.fn();
  const wrapper = shallow(
    <TodoListItem
      todo={defaultTodos[0]}
      index={index}
      handleDelete={handleDelete}
      handleToggle={handleToggle}
    />
  );
  test('should display the component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('should call the handleDelete function', () => {
    wrapper.find('button').simulate('click');
    expect(handleDelete).toHaveBeenCalledWith(defaultTodos[0].id);
  });
  test('should call the handleToggle function', () => {
    wrapper.find('p').simulate('click');
    expect(handleToggle).toHaveBeenCalledWith(defaultTodos[0].id);
  });
  test('should display the paragraph correctly', () => {
    const paragraph = wrapper.find('p').text().trim();
    expect(paragraph).toBe(`${index + 1}. ${defaultTodos[0].desc}`);
  });
  test('should have the class "complete".', () => {
    const todo = defaultTodos[0];
    todo.done = true;
    const wrapper = shallow(
      <TodoListItem
        todo={todo}
        index={index}
        handleDelete={handleDelete}
        handleToggle={handleToggle}
      />
    );
    expect(wrapper.find('p').hasClass('complete')).toBe(true);
  });
});
