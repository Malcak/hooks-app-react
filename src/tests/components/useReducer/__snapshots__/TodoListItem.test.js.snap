// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on the <TodoListItem /> component should display the component correctly 1`] = `
<li
  className="list-group-item"
>
  <p
    className="false"
    onClick={[Function]}
  >
    1
    . 
    Learn react
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    delete
  </button>
</li>
`;
